src/Tools/Pagination/Paginator.php

```diff
@@ -171,13 +428,18 @@ private function cloneQuery(Query $query): Query
     /**
      * Determines whether to use an output walker for the query.
      */
-    private function useOutputWalker(Query $query): bool
+    private function useOutputWalker(Query $query, bool $forCountQuery = false): bool
     {
-        if ($this->useOutputWalkers === null) {
-            return (bool) $query->getHint(Query::HINT_CUSTOM_OUTPUT_WALKER) === false;
+        if ($forCountQuery === false && $this->useResultQueryOutputWalker !== null) {
```

Comment:
`$forCountQuery` is guaranteed to be a boolean, you can use it directly.
```suggestion
        if (! $forCountQuery && $this->useResultQueryOutputWalker !== null) {
```